pkg/spec/resources.go

```diff
@@ -489,6 +490,32 @@ func (app *ControllerFields) CreateK8sObjects() ([]runtime.Object, []string, err
 	objects = append(objects, configMap...)
 	log.Debugf("app: %s, configMap: %s\n", app.Name, spew.Sprint(configMap))
 
+	//Adding Annotations to the metadata of all the objects based on their type
+	if app.Appversion != "" {
+		for i, object := range objects {
+			switch t := object.(type) {
+			case *ext_v1beta1.Ingress:
+				t.ObjectMeta.Annotations = addKeyValueToMap(appVersion, app.Appversion, t.ObjectMeta.Annotations)
+				objects[i] = t
+			case *os_route_v1.Route:
+				t.ObjectMeta.Annotations = addKeyValueToMap(appVersion, app.Appversion, t.ObjectMeta.Annotations)
+				objects[i] = t
+			case *api_v1.Service:
+				t.ObjectMeta.Annotations = addKeyValueToMap(appVersion, app.Appversion, t.ObjectMeta.Annotations)
+				objects[i] = t
+			case *api_v1.Secret:
+				t.ObjectMeta.Annotations = addKeyValueToMap(appVersion, app.Appversion, t.ObjectMeta.Annotations)
+				objects[i] = t
+			case *api_v1.PersistentVolumeClaim:
+				t.ObjectMeta.Annotations = addKeyValueToMap(appVersion, app.Appversion, t.ObjectMeta.Annotations)
+				objects[i] = t
+			case *api_v1.ConfigMap:
+				t.ObjectMeta.Annotations = addKeyValueToMap(appVersion, app.Appversion, t.ObjectMeta.Annotations)
+				objects[i] = t
```

Comment:
@kadel the problem with this approach is that there is no guarantee we might not miss out on new object!

If tomorrow kedge starts supporting a new object and if we miss out on adding that here, we will have same problem as to calling `addKeyValueToMap` in some `fix*` method or in some `create*` method.

Reply:
I think there is a solution to this, if we have a default case where anything that we don't handle will come and we show it with big warning saying that this particular kind was not handled in this type switch!

Reply:
Having default is good practice and should be there! We should look at the solution where we don't even need type switch. Some simple unified way that will add labels to any runtime object regarding the type.